src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -37,12 +44,13 @@ internal void InitializeDefaultAutoCompleteCandidates()
 
         internal void PopulateAutoCompleteCandidates()
         {
-            if(PortViewModel == null) return;
+            if (PortViewModel == null) return;
 
             var searchElements = GetMatchingNodes();
             FilteredResults = searchElements.Select(e =>
             {
-                var vm  = new NodeSearchElementViewModel(e, this);
+                var vm = new NodeSearchElementViewModel(e, this);
+                //TODO I think this leaks.
                 vm.RequestBitmapSource += SearchViewModelRequestBitmapSource;
```

Comment:
Ok, I think its other usage also leaks. I think you can cycle over the `FilteredResults` property of the `SearchViewModel` in its Dispose method and unsubscribe from its `SearchViewModelRequestBitmapSource` method for each `NodeSearchElementViewModel`.

Reply:
unfortunately, as far as I can tell, we only call dispose on the searchViewModel when the workspaceViewModel is disposed, when means that FilteredResults will have been set and reset many times - so we'll only end up calling dispose on the last set of `FilteredResults`, I think this is insufficient.

One day we really need to look into weak references for events...
https://docs.microsoft.com/en-us/dotnet/desktop/wpf/advanced/weak-event-patterns?view=netframeworkdesktop-4.8

I think one approach is to dispose the `FilteredResults` when the view is closed.... I think there might be some subtle differences in allocation between the in canvas search and the node autocomplete though...

I'll tread carefully here, I don't want to effect in canvas search in this PR, we can always handle investigating any leaks as a separate task if that works?

Reply:
How about making `SearchViewModel.SearchViewModelRequestBitmapSource` internal and then doing the following in the `NodeSearchElementViewModel.Dispose` method:
```
if (RequestBitmapSource != null)
{
                    RequestBitmapSource -= searchViewModel.SearchViewModelRequestBitmapSource;
 }
```